compiler/x/env/OMRCPU.cpp

```diff
@@ -64,11 +64,37 @@ OMR::X86::CPU::detect(OMRPortLibrary * const omrPortLib)
 
    if (TRUE == omrsysinfo_processor_has_feature(&processorDescription, OMR_FEATURE_X86_OSXSAVE))
       {
-      static const bool disableAVX = feGetEnv("TR_DisableAVX") != NULL;
-      if (((6 & _xgetbv(0)) != 6) || disableAVX) // '6' = mask for XCR0[2:1]='11b' (XMM state and YMM state are enabled)
+      // Check XCRO register for OS support of xmm/ymm/zmm
+      // '6' = mask for XCR0[2:1]='11b' (XMM state and YMM state are enabled)
+      static const bool disableAVX = ((6 & _xgetbv(0)) != 6) || feGetEnv("TR_DisableAVX") != NULL;
+      static const bool disableAVX2 = disableAVX || feGetEnv("TR_DisableAVX2") != NULL;
+      // 'e0' = mask for XCR0[7:5]='111b' (Opmask, ZMM_Hi256, Hi16_ZMM)
+      static const bool disableAVX512 = ((0xe0 & _xgetbv(0)) != 0xe0) || disableAVX || disableAVX2 || feGetEnv("TR_DisableAVX512") != NULL;
```

Comment:
I realize you checked for XCR0[2:1] on L69 which makes it unnecessary here, but for someone reading the Intel spec and expecting to find those bits checked in the AVX512 case they'll have to study this code to grok this fact.

It costs nothing to re-check them here, so for readability can you change the mask to `0xe6` (I think)?  Unless there is some other reason you didn't do that, in which case please enlighten.

Ditto for the duplicate code in the JIT.

Reply:
Thanks for making the change, but is there a reason you didn't update this mask?